pkg/util/iptables/save_restore.go

```diff
@@ -55,3 +56,40 @@ func GetChainsFromTable(save []byte) sets.Set[Chain] {
 	}
 	return chainsSet
 }
+
+// GetCountersForRule parse and returns packet and byte counters associated with the rule.
+func GetCountersForRule(save []byte, rule string) (int, int, error) {
+	// \[(\d+):(\d+)] -A KUBE-FORWARD -m conntrack --ctstate INVALID -j DROP
+	var err error
+	index := bytes.Index(save, []byte(rule))
+	pCounter, bCounter := 0, 0
+	if index == -1 {
+		return pCounter, bCounter, fmt.Errorf("rule not found")
+	}
```

Comment:
you just do "return 0, 0, ..." in the other error cases (which would then let you move the "rule" check before the pCounter, bCounter declaration)